src/cargo/core/resolver/mod.rs

```diff
@@ -681,11 +695,13 @@ fn activate_deps_loop<'a>(mut cx: Context<'a>,
                         parent: Summary::clone(&parent),
                         dep: Dependency::clone(&dep),
                         features: Rc::clone(&features),
+                        conflicting_activations: conflicting_activations.clone(),
```

Comment:
I'm 90% sure this is always empty - the only place we add to `conflicting_activations` is when we've run out of candidates, in which case we're about to backtrack...which means `conflicting_activations` immediately gets overwritten with an empty set again.

This makes sense, because when backtracking you only care about the conflicts for the current dep rather than for the whole resolve history. It also simplifies the code significantly.

Reply:
Indeed, `conflicting_activations` is at this time mostly unused. It is used to move `conflicting` to `find_candidate` and then use to shuttle the info to `activation_error`. The intent is to allow anything to add to the `conflicting_activations` at any point in the proses. (specifically for #5000). It would be reasonable to remove it in this PR and add it back in in that PR iif it turns out to be useful there.

Reply:
@Eh2406 that sounds like a good strategy to me!

Reply:
Done.